src/modules/kvs/kvs.c

```diff
@@ -1379,12 +1380,21 @@ static void getroot_request_cb (flux_t *h, flux_msg_handler_t *mh,
                              "namespace", &namespace) < 0)
         namespace = KVS_PRIMARY_NAMESPACE;
 
-    /* If root is not initialized, we have to intialize ourselves
-     * first.
-     */
-    if (!(root = getroot (ctx, namespace))) {
-        flux_log_error (h, "getroot");
-        goto error;
+    if (ctx->rank == 0) {
+        if (!(root = zhash_lookup (ctx->roothash, namespace))) {
+            flux_log (h, LOG_INFO, "namespace %s not found", namespace);
+            errno = EINVAL;
```

Comment:
ENOENT might be more appropriate here?

Reply:
I had debated ENOENT vs EINVAL.  Admittedly ENOENT feels the right one, but I sided with EINVAL b/c if a user does "flux kvs get foo.bar" and gets a ENOENT, the initial thought would be that "foo.bar" doesn't exist, instead of the namespace doesn't exist.  But EINVAL is confusing as well, perhaps ENOENT would be slightly better.

Reply:
@chu11, you make a good point. And that makes me worry about the long-term usability of relying solely on system errno as a way to return errors to the end user.

Perhaps for now we have to resort to custom errno's to determine the difference between an namespace that doesn't exist vs a path in that namespace that doesn't exist.

Or, how about something like ENOTSUP (Literally operation not supported), which `flux-kvs` could decode as namespace doesn't exist (as differentiated from key doesn't exist ENOENT and invalid parameter EINVAL)?

Reply:
@grondo Ya know, I didn't see ENOTSUP.  It's not the standard errno headers on catalyst.  But atleast one header seems to map to EOPNOTSUPP, which is listed as "/* Operation not supported on transport endpoint */".  Not perfect, but is close.

Reply:
Hmmm.  Some watch code assumes ENOENT means "foo.bar" is missing and will wait on it.  So determining "bad namespace" vs "key missing" isn't possible.

Should a kvs-watch work with namespaces that don't yet exist?  Don't think it's too hard to do, but not sure I want to do in this PR.  Perhaps should keep EINVAL until we do this?

Reply:
Hmm, if we want we could add a libflux/errno.[ch] that defines a custom errno range and matching array of strings.  We are already assuming that RPC errnum responses can contain zeromq errors, which have their own numerical range.  And `flux_strerror()` calls `zmq_strerror()` to decode them.   The idea was always that `flux_strerror()` could be short circuited to decode a custom flux-core range as well.

I think we haven't acted on that because we had a better idea in #796.  However, since that hasn't happened, and the KVS is a pretty fundamental service, I wouldn't feel too bad if we went ahead and added some KVS specific errors that could be decoded by `flux_strerror()`.

Otherwise, @grondo's suggestion of co-opting another errno is maybe OK.

Regarding kvs watch, I think it's fine to have them fail if the namespace doesn't exist yet.

Reply:
Ahh, wasn't aware that we called ```zmq_strerror()``` internally.  So using ENOTSUP makes sense.